google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/facet/AppEngineStandardFacet.java

```diff
@@ -52,4 +56,51 @@ public static AppEngineStandardFacet getAppEngineFacetByModule(@Nullable Module
     return FacetManager.getInstance(module).getFacetByType(ID);
   }
 
+  /**
+   * Returns {@code true} if this app is targeting the Flexible Environment instead of the Standard
+   * Environment.
+   */
+  public boolean isFlexEnvironment() {
+    AppEngineStandardWebApp appEngineStandardWebApp = getAppEngineStandardWebXml();
+    return appEngineStandardWebApp != null
+        && "flex".equalsIgnoreCase(appEngineStandardWebApp.getEnv().getStringValue());
+  }
+
+  /**
+   * Returns {@code true} if this app is targeting the Managed VM environment instead of the
+   * Standard Environment.
+   */
+  public boolean isManagedVmEnvironment() {
+    AppEngineStandardWebApp appEngineStandardWebApp = getAppEngineStandardWebXml();
+    return appEngineStandardWebApp != null
+        && "true".equalsIgnoreCase(appEngineStandardWebApp.getVm().getStringValue());
+  }
+
+  /**
+   * Returns {@code true} if this app is targeting some environment other than the Standard
+   * Environment.
+   */
+  public boolean isNonStandardEnvironment() {
```

Comment:
@etanshaul funny you should ask.  I thought about this for a bit and my two rationales for doing it this way is that the is an AppEngineStandardFacet so I kind of assume that it's a standard environment.  The higher-order question I want to know is whether it's targeting some unusual non-standard environment, and since those environments are eventually going away (for standard) I can just remove the code paths that take the non-standard routes later.

It's a bit metaphysical I grant you. So I'm not deeply entrenched to this approach.